Public roadmap

mapsam opened the pull request:
This is a first step towards a beautiful, helpful, public roadmap #3559 

<img width="807" alt="screen shot 2016-11-11 at 11 12 56 am" src="https://cloud.githubusercontent.com/assets/1943001/20227121/d7d9f7b2-a7ff-11e6-8a23-94d6887b250a.png">

I've gone through a _very rough_ first draft of the roadmap, but my perspective is limited and it needs input from everyone else! I'll plan on manually updating this during weekly scrum. If anyone has extra features they think would be helpful, let me know and I can add them.

cc @lucaswoj @mollymerp @lbud @mourner @jfirebaugh @ansis

mapsam commented:
Good idea @mourner! That will keep the workflow similar to the `plugins.yml`.

mollymerp commented:
thank you for taking initiative on this @mapsam ! great work!

mapsam commented:
@mourner added a `roadmap.yml` file and simplified the layout.

lucaswoj commented:
I'm stoked about this 🙌 

Here are my notes about what the _contents_ of the roadmap should include: 

# Roadmap

We don’t have a roadmap yet so you get some discretion in populating it. Filling in the roadmap will have to be an iterative process as different stakeholders speak up. 

## What should go on the roadmap

* Everything in “Project ⛈” 
* Everything sitting in an open PR is top priority right now
* Everything labeled “high priority” is pretty high priority
* Everything labeled “breaking change” is pretty high priority (we wanna v1 as soon as we can)

## What should not go on the roadmap

* small bug fixes
* small features (~1 days of work)
* refactoring projects

mapsam commented:
Per discussion with @lucaswoj I've updated the content and added the ability to link an item to a project on Github. This is ready to go whenever we're comfortable!

![image](https://cloud.githubusercontent.com/assets/1943001/20234581/8e1386ba-a832-11e6-8df0-1cf65b6af551.png)

mapsam commented:
@lucaswoj thanks for posting those notes.

> [do not include] refactoring projects

I'm a little hesitant to completely cut these out, especially if we know they are coming. It might be helpful for people to see these changes coming on the horizon and prepare.

lucaswoj commented:
Fair point re: refactoring. We should handle this on a case-by-case basis.

lucaswoj commented on docs/_data/roadmap.yml:
```diff
@@ -0,0 +1,33 @@
+      name: Version 1.0 and API stability
+      issues: ["2803", "2727", "2506", "2112", "1408"]
+    -
+      name: Custom shaders
```
Let's bump this off the roadmap while we see how #3580 pans out

mapsam replied:
👍

mapsam replied:
I removed "custom shaders" ... do we want #3580 added @lucaswoj?

lucaswoj commented on docs/_data/roadmap.yml:
```diff
@@ -0,0 +1,33 @@
+      name: Custom source type API
+      project: 2
+-
+  term: "Long-term"
```
I'm unsure how to determine 

 - if a particular issue should be in `upcoming` or `long term`
 - when a particular issue should be moved from `long term` to `upcoming`

Do you have thoughts on how we can make these ☝️ determinations?  Do you think it'd be easier to manage if we had only two pools of issues: `active development` and `upcoming`?

mapsam replied:
I think it's a great idea to just have two columns to start with. We can add a third if it feels necessary during Tuesday scrum.

lucaswoj commented on docs/_data/roadmap.yml:
```diff
@@ -0,0 +1,33 @@
+- 
```
We could put some procedural notes at the top of this file in a comment. Some points we should hit:

 - This roadmap is updated every week during the Tuesday scrum
 - This roadmap should _not_ include...
  * small bug fixes
  * small features (~1 day of work)
 - This roadmap should not have more than N items in the "Upcoming" section at any time (10?)

mapsam commented on docs/_data/roadmap.yml:
```diff
@@ -0,0 +1,33 @@
+      issues: ["2803", "2727", "2506", "2112", "1408"]
+    -
+      name: Custom shaders
+      issues: ["281"]
```
Added!

mapsam commented:
Thanks for the comments @lucaswoj - I've updated the page to be two columns, added some additional procedural text, and updated the roadmap items. Screenshot:

![screen shot 2016-11-15 at 8 21 04 am](https://cloud.githubusercontent.com/assets/1943001/20313772/ca88a24c-ab0c-11e6-8fac-71d8b4e68a0d.png)

mapsam commented:
Wahoo!
